wordpress-mobile/WordPress-FluxC-Android
# [Build] Enable Non-Transitive Resources

This PR is a prerequisite for the `Gradle 8.1.1 & AGP 8.0.2 Upgrade for WPAndroid, WCAndroid & Related Libs` project.

Platform Request: `pdnsEh-13V-p2`
Project Thread: `paaHJt-57Z-p2`

-----

This PR enables non-transitive resources (`android.nonTransitiveRClass`) for the project.

FYI: This behavior becomes the default in AGP `8.0` and higher. As such, this becomes a prerequisite for the AGP `8.0.2` upgrade, that is of course, unless `android.nonTransitiveRClass` is explicitly set to `false`.

PS: This PR also disables Jetifier (87139a60e0528b3e7d18ca39beff64b07cee0968), something that was [already discussed](https://github.com/wordpress-mobile/WordPress-Android/pull/15675#issuecomment-1057840724) back then (1,5 years ago), but we never got around to applying and testing this change.

-----

## To test:

1. Verify that all the CI checks are successful.
2. Smoke test the app.
3. Also, because of this (87139a60e0528b3e7d18ca39beff64b07cee0968) and this (251b1a5cdbf4caf76171b529db5c79bb67c70b50) changes, just to be on the safe side, you could quickly smoke test both, the WPAndroid and WCAndroid apps, and verify that everything is working as expected.

FYI: Due to the above 87139a60e0528b3e7d18ca39beff64b07cee0968 and 251b1a5cdbf4caf76171b529db5c79bb67c70b50 changes, I am also considering updating the [wellsql](https://github.com/wordpress-mobile/wellsql) project and level-it-up with all other repos that we are currently supporting. Maybe then, we can drop the `com.android.support` group dependency exclusion from the `wellsql` dependency altogether, and this way, to avoid any future bottlenecks when working with [wellsql](https://github.com/wordpress-mobile/wellsql). Wdyt @oguzkocer ?

-----

Please also notice this 8fff0be773d82b938d46d00d08b7eb476720ffcf and its description. This commit doesn't updates the encrypted `gradle.properties.enc` file as it seems that this kind of encrypted `gradle.properties` file isn't available for this project.

Can you please confirm the above? 🤔 

And if that's the case, let's take this opportunity to remove any such encryption related functionality from this repo.
